homeassistant/components/hunterdouglas_powerview/button.py

```diff
@@ -0,0 +1,135 @@
+"""Buttons for Hunter Douglas Powerview advanced features."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from typing import Any, Final
+
+from aiopvapi.resources.shade import BaseShade, factory as PvShade
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    COORDINATOR,
+    DEVICE_INFO,
+    DOMAIN,
+    PV_API,
+    PV_ROOM_DATA,
+    PV_SHADE_DATA,
+    ROOM_ID_IN_SHADE,
+    ROOM_NAME_UNICODE,
+)
+from .coordinator import PowerviewShadeUpdateCoordinator
+from .entity import ShadeEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class PowerviewButtonDescriptionMixin:
+    """Mixin to describe a Button entity."""
+
+    press_action: Callable[[BaseShade, PowerviewButton], Any]
+
+
+@dataclass
+class PowerviewButtonDescription(
+    ButtonEntityDescription, PowerviewButtonDescriptionMixin
+):
+    """Class to describe a Button entity."""
+
+
+BUTTONS: Final = [
+    PowerviewButtonDescription(
+        key="calibrate",
+        name="Calibrate",
+        icon="mdi:swap-vertical-circle-outline",
+        device_class=ButtonDeviceClass.UPDATE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        press_action=lambda shade, button: shade.calibrate(),
+    ),
+    PowerviewButtonDescription(
+        key="identify",
+        name="Identify",
+        icon="mdi:crosshairs-question",
+        device_class=ButtonDeviceClass.UPDATE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        press_action=lambda shade, button: shade.jog(),
+    ),
+    PowerviewButtonDescription(
+        key="update",
+        name="Force Update",
+        icon="mdi:autorenew",
+        device_class=ButtonDeviceClass.UPDATE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        press_action=lambda shade, button: button.async_update(),
+    ),
```

Comment:
We should drop the button as well as using the built-in service is the preferred way

Reply:
Its won't poll automatically since it uses CoordinatorEntity as a base class.  They will have to call the service to the individual shade to poll

https://github.com/home-assistant/core/blob/95e9bd106eb2a7077e92aa7c8653d668ebe09533/homeassistant/helpers/update_coordinator.py#L317

Reply:
sure thats fine then, ill remove the button - but having now updated the asyc_update for this item wouldnt that mean battery shades would be getting spammed with a force refresh every 60 seconds ?

Or is async_update only called through service and not related to the 60 second refresh ?

Reply:
`async_update` is called automatically when `should_poll` is `True` on the interval based on the `UPDATE_INTERVAL` constants for the platform.  Since the `CoordinatorEntity`, which is the base call for all of these entities, sets `should_poll` to `False` it will not poll automatically and only get called when they call the `homeassistant.update_entity` service (or one of the other standard core methods which probably aren't relevant here)